src/infra/docs/rustc-ci.md

```diff
@@ -31,14 +38,20 @@ platform’s custom Docker container. This has a lot of advantages for us:
   time thanks to Docker image caching.
 - Users can run the same tests in the same environment locally by just running
   `src/ci/docker/run.sh image-name`, which is awesome to debug failures.
+- Forces the use of portable scripts to drive the CI process which keeps the CI fairly platform independent (i.e., we are not overly reliant on GitHub Actions).
```

Comment:
I'm not sure if this is true. We could've also built our CI with portable scripts that run inside a VM.

I might move this into its own paragraph and add another sentence or two explaining that portability is an explicit design goal. How we achieve portability is to rely on shell scripts and Docker, and use GitHub Actions only for orchestration.

I.e. most/all (?) steps on GitHub Actions call a script, instead of running a command directly:

```yaml
- run: build.sh
```

vs.

```yaml
- run: cargo build
```

That's at least my understanding.

Reply:
Correct, everywhere we can we invoke shell scripts. The only exception is running code before cloning the repository.